src/app/shared/models/providerAdmin.model.ts

```diff
@@ -13,3 +14,27 @@ export interface ProviderAdminTable {
   phone: string,
   deputy: string,
 }
+
+export class ProviderAdmin {
+  id?: string;
+  userId?: string;
+  firstName: string;
+  middleName: string;
+  lastName: string;
+  email: string;
+  phoneNumber: string;
+  providerId: string;
+  isDeputy: string;
+  managedWorkshopIds?: Array<string>;
+
+  constructor(info, providerId, isDeputy, workshopIds?: Array<string>) {
+    this.email = info.email;
+    this.phoneNumber = info.phoneNumber;
+    this.firstName = info.firstName;
+    this.middleName = info.middleName;
+    this.lastName = info.lastName;
+    this.providerId = providerId;
+    this.isDeputy = isDeputy;
+    this.managedWorkshopIds = (workshopIds) ? workshopIds : [];
```

Comment:
check this logic one more time please. You sure the  workshopIds could be passed as undefined/null? Is it a required field (Maybe we can skip assigning an empty array here). If it is still necessary, you can provide a default value to the parameter.

Reply:
Skipped assigning an empty array here. It is really not necessary.